File: typescript/interfaceTest.ts

interface Sayer{
	name:String
	age:Number
	say:() => void
}

function test() {
	var a:Sayer = {
		name: "Stanley",
		age: 20,
		say: () => {console.log(a.name, a.age)}
	}
	a.say()
}

test()

File: kotlin/interfaceTest.kt

interface Sayer{
    fun say()
    fun say(words:String){
        println(words)
    }
}

class Person(var name:String, var age:Int):Sayer{
    
    override fun say(){
        println("${this.name} ${this.age}")
    }
}


fun test(){
    var a = Person("Stanley", 20)
    a.say()
    a.say("apple")
}